### PR TITLE
Service cache docs

### PR DESCRIPTION
**Pull request by ludwikbukowski**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6722.1](https://travis-ci.org/esl/MongooseIM/jobs/550230841) / Erlang 20.3 / small_tests / 80081a41a66253c83902d5a9c7a151ac1ab994ee
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2347/6722/small_tests.20.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2347/6722/small_tests.20.3/small/ct_run.mongooseim@localhost.2019-06-25_11.15.40/lib.mongooseim.logs/run.2019-06-25_11.15.41/suite.log.html)

**Comment by mongoose-im** [comment]
[6723.1](https://travis-ci.org/esl/MongooseIM/jobs/550231727) / Erlang 20.3 / small_tests / 6861bf02beefa99318ca0d64fd12110f7f397e39
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2347/6723/small_tests.20.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2347/6723/small_tests.20.3/small/ct_run.mongooseim@localhost.2019-06-25_11.53.40/lib.mongooseim.logs/run.2019-06-25_11.53.40/suite.log.html)

File: doc/advanced-configuration/Services.md [hunk 1]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+    * `gdpr`: Adds `retrieve_personal_data`
 
 ### Example configuration
 ` {service_admin_extra, [{submods, [node, accounts, sessions]}]} `
+
+Module `service_admin_extra` depens on `service_cache`(see below) as cache is efficiently used for `gdpr` submodule and GDPR compilant account removal.
+
+### service_cache
+
+Implementation of cache storage.
```
Comment (mattsaidwhat) [inline doc/advanced-configuration/Services.md]: The implementation

File: doc/advanced-configuration/Services.md [hunk 2]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+Module expose one function `lookup/2` with parameters `Key` and `Fun`.
```
Comment (mattsaidwhat) [inline doc/advanced-configuration/Services.md]: module exposes one

File: doc/advanced-configuration/Services.md [hunk 3]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+When called and the key `Key` exists in cache, the related value is returned.
```
Comment (mattsaidwhat) [inline doc/advanced-configuration/Services.md]: when the function is called

File: doc/advanced-configuration/Services.md [hunk 4]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
     * `stanza`: Adds `send_message_chat`, `send_message_headline`, `send_stanza_c2s`
     * `stats`: Adds `stats`, `stats_host`
     * `vcard`: Adds `get_vcard`, `get_vcard2`, `get_vcard2_multi`, `set_vcard`, `set_vcard2`, `set_vcard2_multi`
+    * `gdpr`: Adds `retrieve_personal_data`
 
 ### Example configuration
 ` {service_admin_extra, [{submods, [node, accounts, sessions]}]} `
+
+Module `service_admin_extra` depens on `service_cache`(see below) as cache is efficiently used for `gdpr` submodule and GDPR compilant account removal.
```
Comment (fenek) [inline doc/advanced-configuration/Services.md]: depens -> depends

Comment (fenek) [reply doc/advanced-configuration/Services.md]: `as the cache speeds up GDPR data retrieval and removal.`?

File: doc/advanced-configuration/Services.md [hunk 5]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+
+### service_cache
+
+The Implementation of cache storage.
```
Comment (fenek) [inline doc/advanced-configuration/Services.md]: Implementation -> implementation

**Comment by mongoose-im** [comment]
[6728.1](https://travis-ci.org/esl/MongooseIM/jobs/550266805) / Erlang 20.3 / small_tests / 3e406bdfad9fefb9e016dcf9cb359833f4991721
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2347/6728/small_tests.20.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2347/6728/small_tests.20.3/small/ct_run.mongooseim@localhost.2019-06-25_13.26.22/lib.mongooseim.logs/run.2019-06-25_13.26.23/suite.log.html)

File: doc/advanced-configuration/Services.md [hunk 6]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
     * `stanza`: Adds `send_message_chat`, `send_message_headline`, `send_stanza_c2s`
     * `stats`: Adds `stats`, `stats_host`
     * `vcard`: Adds `get_vcard`, `get_vcard2`, `get_vcard2_multi`, `set_vcard`, `set_vcard2`, `set_vcard2_multi`
+    * `gdpr`: Adds `retrieve_personal_data`
 
 ### Example configuration
 ` {service_admin_extra, [{submods, [node, accounts, sessions]}]} `
+
+Module `service_admin_extra` depends on `service_cache`(see below) as the cache speeds up GDPR data retrieval and removal.
```
Comment (goddammit) [inline doc/advanced-configuration/Services.md]:
```suggestion
The `service_admin_extra` module depends on the`service_cache`(see below) as the cache speeds up GDPR data retrieval and removal.
```

File: doc/advanced-configuration/Services.md [hunk 7]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+
+### service_cache
+
+The implementation of cache storage.
```
Comment (goddammit) [inline doc/advanced-configuration/Services.md]:
```suggestion
Implementation of the cache storage.
```
not sure which one sounds better

File: doc/advanced-configuration/Services.md [hunk 8]
```diff
@@ -29,6 +29,28 @@ Provides additional commands to mongooseimctl script.
...
+Module exposes one function `lookup/2` with parameters `Key` and `Fun`.
```
Comment (goddammit) [inline doc/advanced-configuration/Services.md]:
```suggestion
The module exposes a function `lookup/2` with parameters `Key` and `Fun`.
```

**Pull request merged**